### ISSUE TITLE
FIRAnalytics setConsent request hangs when `FIRConsentTypeAdStorage` is `FIRConsentStatusDenied`

### ISSUE THREAD

**Issue by russellwheatley** [firebase/firebase-ios-sdk#8949]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.1
  * Firebase SDK version: 8.9.0
  * Installation method: `CocoaPods` (select one)
  * Firebase Component: Analytics 

### [REQUIRED] Step 2: Describe the problem

Try to set `FIRConsentTypeAdStorage` to `FIRConsentStatusDenied` in the `setConsent` API will cause the request to hang.

#### Steps to reproduce:

Simply run the below noted `Relevant Code` and you ought to see the request hang. 
#### Relevant Code:

```obj-c
NSMutableDictionary<FIRConsentType, FIRConsentStatus> *parameters = [[NSMutableDictionary alloc] init];
parameters[FIRConsentTypeAdStorage] = FIRConsentStatusDenied;
[FIRAnalytics setConsent:parameters];
```

**Comment by rizafran** [comment]
Hi @russellwheatley, it looks like I was able to reproduce the issue. To make sure that we're on the same page, is the request hanging only upon opening the app and is not reproducible if using `FIRConsentStatusGranted`?

**Comment by russellwheatley** [comment]
Hi @rizafran, The request is fine if you use `FIRConsentStatusGranted`. The request never completes if you use `FIRConsentStatusDenied `. I'm afraid I don't have it open to test the specific timing of when the request fails.

**Comment by rizafran** [comment]
Thanks for confirming, @russellwheatley. It looks like we're having the same behavior. With this, I'll notify an engineer and see what we can do here.

**Comment by tsunghung** [comment]
Thanks for the bug report. We found a deadlock in the Analytics SDK during initialization. We will fix it in the next release.
In the meantime, to disable Ad storage by default, you can set the value of `GOOGLE_ANALYTICS_DEFAULT_ALLOW_AD_STORAGE` to `NO` in your app's Info.plist file. For example, viewed in the source XML:
```
<key>GOOGLE_ANALYTICS_DEFAULT_ALLOW_AD_STORAGE</key>
<false/>
```
You can still use the `setConsent` API to change the consent status afterward.

**Comment by paulb777** [comment]
This fix is now targeted for the release after next - 8.12.0

**Comment by eddie8balltoliver** [comment]
> Thanks for confirming, @russellwheatley. It looks like we're having the same behavior. With this, I'll notify an engineer and see what we can do here.

**Comment by ryanwilson** [comment]
This was published and in Firebase pod 8.12.1 (FirebaseAnalytics pod version 8.12.0).
